modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -1307,7 +1307,8 @@ private void _validateAccountEntryRestrictedObjectFieldId(
 		}
 
 		if (objectDefinition.isApproved() &&
-			objectDefinition.isAccountEntryRestricted()) {
+			(objectDefinition.isAccountEntryRestricted() !=
+				accountEntryRestricted)) {
```

Comment:
Hi @TamyrisTorres I suspect, by reading the error message below `Account entry restriction cannot be disabled when the object definition is published`, that the right condition here is:

```
objectDefinition.isApproved() && 
      objectDefinition.isAccountEntryRestricted() && 
            !accountEntryRestricted
``` 

If `objectDefinition.isAccountEntryRestricted()` is `true` then it cannot be changed to `false`.

Could you please check?

Reply:
@carolmariaabb I made the change in the message as the customer's need is that we can update the accountEntryRestricted when it has not been published.

Reply:
> as the customer's need is that we can update the accountEntryRestricted when it has not been published.

@TamyrisTorres did you confirm with the product team if this is the expected behavior?

Reply:
Hi @carolmariaabb. Yes, I talked about it with Felipe Marinho and we improved the code. Please review the code one more time. Thanks.